javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -91,11 +91,10 @@ private static void includeEmbeddedExtensionsIfFound(
           File tempFile = new File(tempDirectory, name.substring(prefix.length()));
           // reject extensions that would be extracted outside of temp directory
           // https://security.snyk.io/research/zip-slip-vulnerability
-          if (name.indexOf("..") != -1
-              && !tempFile
-                  .getCanonicalFile()
-                  .toPath()
-                  .startsWith(tempDirectory.getCanonicalFile().toPath())) {
+          if (!tempFile
+              .getCanonicalFile()
+              .toPath()
```

Comment:
```suggestion
              .getCanonicalPath()
```

Reply:
These to are stubbly different and using just `getCanonicalPath()` would be wrong. `getCanonicalPath()` returns a `String` and as you noted yourself when comparing with `startsWith` you would need to add `/` to ensure the comparison works as intended. In contrast when `toPath()` is used we'll be using `startsWith` on a `Path` not `String` that already does the comparison by segment. In my opinion the current approach is better.